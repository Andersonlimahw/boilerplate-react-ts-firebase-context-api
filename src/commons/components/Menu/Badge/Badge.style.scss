.badge {
    display: flex;
    color: #121212;
    border-radius: 32px;
    background: #f1f1f1;
    padding:  0 8px;
    font-weight: 700;
    font-size: 0.75rem;
    line-height: 1.5rem;
    border: 1px solid #f1f1f1;

    &.default {
        color: #121212;
    }

    &.warning {
        background: #F4B400;
        border-color: #F4B400;
    }

    &.danger {
        background: #DB4437;
        border-color: #DB4437;
    }
}