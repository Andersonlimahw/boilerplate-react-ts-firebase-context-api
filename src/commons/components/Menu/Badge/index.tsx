import { FC, ReactNode } from "react";

import './Badge.style.scss';

interface BadgeProps {
    variant: 'default' | 'warning' | 'danger',
    children: ReactNode | ReactNode[],
}
export const Badge: FC<BadgeProps> = ({ variant, children }) => {
    return (<div className={`badge ${variant}`}> {children} </div>);
}

export default Badge;