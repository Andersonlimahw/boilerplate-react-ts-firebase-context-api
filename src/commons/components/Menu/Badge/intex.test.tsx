import React from 'react';
import { render } from '@testing-library/react';
import Badge from './';

describe('Badge', () => {
    test('renders with default variant', () => {
        const { getByText, asFragment } = render(<Badge variant="default">Default</Badge>);
        const badge = getByText(/Default/i);
        expect(badge).toBeTruthy();
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders with warning variant', () => {
        const { getByText, asFragment } = render(<Badge variant="warning">Warning</Badge>);
        const badge = getByText(/Warning/i);
        expect(badge).toBeTruthy();
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders with danger variant', () => {
        const { getByText, asFragment } = render(<Badge variant="danger">Danger</Badge>);
        const badge = getByText(/Danger/i);
        expect(badge).toBeTruthy();
        expect(asFragment()).toMatchSnapshot();
    });
});