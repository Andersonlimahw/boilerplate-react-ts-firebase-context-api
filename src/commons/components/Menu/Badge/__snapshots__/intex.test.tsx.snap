// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders with danger variant 1`] = `
<DocumentFragment>
  <div
    class="badge danger"
  >
     Danger 
  </div>
</DocumentFragment>
`;

exports[`Badge renders with default variant 1`] = `
<DocumentFragment>
  <div
    class="badge default"
  >
     Default 
  </div>
</DocumentFragment>
`;

exports[`Badge renders with warning variant 1`] = `
<DocumentFragment>
  <div
    class="badge warning"
  >
     Warning 
  </div>
</DocumentFragment>
`;
